apis/v1alpha1/shared_types.go

```diff
@@ -181,24 +181,27 @@ type RouteGatewayStatus struct {
 	//
 	// +listType=map
 	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=8
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	Conditions []metav1.Condition `json:"conditions"`
 }
 
 // RouteStatus defines the observed state that is required across
 // all route types.
 type RouteStatus struct {
-	// Gateways is a list of the Gateways that are associated with the
-	// route, and the status of the route with respect to each of these
-	// Gateways. When a Gateway selects this route, the controller that
-	// manages the Gateway should add an entry to this list when the
-	// controller first sees the route and should update the entry as
-	// appropriate when the route is modified.
-	//
-	// A maximum of 100 Gateways will be represented in this list. If this list
-	// is full, there may be additional Gateways using this Route that are not
-	// included in the list.
-	//
+	// Gateways is a list of Gateways that are associated with the route,
+	// and the status of the route with respect to each Gateway. When a
+	// Gateway selects this route, the controller that manages the Gateway
+	// must add an entry to this list when the controller first sees the
+	// route and should update the entry as appropriate when the route is
+	// modified.
+	//
+	// A minimum of 1 and a maximum of 100 Gateways will be represented
```

Comment:
I don't think we can require a minimum at this level. A Route may not be selected by any Gateways.

Reply:
@robscott if that's the case, then I would expect `status.gateways` to be optional or `minItems=1` with a default that includes a nil gatewayRef and a default status condition. For example:
```
status:
  gateways:
  - conditions:
     - type: Admitted 
        status: False
        message: Waiting for gateway
        reason: waiting

Reply:
I think the ideal empty state here is actually `gateways: []`. It does seem helpful to have some indication that no Gateways are using this route. With that said, I'm not sure if we could accomplish that though. Maybe a default `[]` combined with no `omitempty` would work? Or maybe this could be a "required" field with a min length of 0? 

I believe the example you have above would be invalid because GatewayRef is a required field. In my opinion, it does not make sense to make GatewayRef optional in a field that is all about describing the relationship of a route with a specific Gateway. If we did that, would a controller need to remove that when they were adding their own status? Would the controller need to add the empty status back if they were removing their entry in this list?

Reply:
> I believe the example you have above would be invalid because GatewayRef is a required field.

What if we default GatewayRef to "None". For example:
```
status:
  gateways:
  - gatewayRef:
     name: None
     namespace: None
    - conditions:
       - type: Admitted 
          status: False
          message: Waiting for gateway
          reason: waiting
```

an owning controller is responsible for updating the condition.